### PR TITLE
#298 Implemented GitlabOrganizationRepos.

### PR DESCRIPTION
**Pull request by criske**
- implemented and tested.
- left puzzle for gitlab invitations.

PR for #298

### CODE REVIEW

File: self-core-impl/src/main/java/com/selfxdsd/core/GitlabOrganizationRepos.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     * @param owner Current authenticated User.
+     * @param resources Gitlab's JSON Resources.
+     * @param storage Storage used by Organization Repo.
+     */
+    GitlabOrganizationRepos(final String organizationId,
+                            final User owner,
+                            final JsonResources resources,
+                            final Storage storage) {
+        this.uri = URI.create("https://gitlab.com/api/v4/groups/"
+            + organizationId + "/projects?simple=true&owned=true");
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/GitlabOrganizationRepos.java:56]: @criske the URI should be constructed on the fly, starting from ``GitLab`` and going down.
See how Github is implemented :D 

Or is there a special reason why you did it like this?

File: self-core-impl/src/main/java/com/selfxdsd/core/GitlabRepo.java [hunk 2]
```diff
@@ -54,6 +54,24 @@
...
+     * @return GitlabRepo.
+     */
+    static GitlabRepo createFromName(final String repoName,
+                                     final JsonResources resources,
+                                     final User owner,
+                                     final Storage storage) {
+        final URI repo = URI.create("https://gitlab.com/api/v4/projects/"
+            + repoName);
+        return new GitlabRepo(resources, repo, owner, storage);
+    }
```
Comment (amihaiemil) [inline self-core-impl/src/main/java/com/selfxdsd/core/GitlabRepo.java:73]: @criske Why is this method needed?

File: self-core-impl/src/main/java/com/selfxdsd/core/GitlabOrganizationRepos.java (hunk 1 above)
Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/GitlabOrganizationRepos.java:56]: Is easier to test. For Github makes sense cause there we dealing with org repos uri extracted from json.
https://github.com/self-xdsd/self-core/blob/dd2648290b4baad63189299c60deb8f2e33daac2/self-core-impl/src/test/java/com/selfxdsd/core/GithubOrganizationTestCase.java#L53
But here we only need the org `id` and with this aproach I'll avoid testing the uri twice: in` GitlabOrganizationTestCase` and in `GithubOrganizationReposTestCase`; But only once see: `GitlabOrganizationReposTestCase#fetchesAndFilterOk` in PR

File: self-core-impl/src/main/java/com/selfxdsd/core/GitlabRepo.java (hunk 2 above)
Comment (criske) [reply self-core-impl/src/main/java/com/selfxdsd/core/GitlabRepo.java:73]: Same logic for uri creation is used in different places (`Gitlab` and `GitlabOrganizationRepos`). But the main inconvenient is that Gitlab API doesn't give full api-url for repos in their json responses as Github does. So each time we need to extract the repo name from json's `path`-key  and from there build the Uri used by `GitlabRepo`.

**Comment by amihaiemil** [comment]
@criske I see, ok.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Pull request merged**